Automattic/jetpack
# wp-admin Site Default: Fix/enable the menu toggle when in wp-admin

Fixes: https://github.com/Automattic/dotcom-forge/issues/4742

The sidebar can not be toggled by clicking on the My Sites button on a mobile view (<=782 pixels). In Calypso, clicking on the My Sites button while on a mobile display will trigger the sidebar but due to the masterbar adjustments, the toggle is not working in wp-admin with the masterbar module active. This is because the original WordPress toggle is replaced with the My Sites button, but no toggle functionality exists for it.

## Proposed changes:


This PR turns the 'My Sites' button in a sidebar toggle while on a mobile display.

* Clicking on the 'My Sites' admin bar button while on a display that is <= 782 pixels toggles the sidebar.
* Clicking outside of the sidebar while it is opened and on a display that is <= 782 pixels will hide the sidebar.
* Disables the default link behavior while on a smaller viewport size so it becomes a toggle.
* Removes conflicting event handlers from the original toggle element.
* Prevents the click event from propagating when on a mobile viewport size so it does not trigger the original behavior of `.ab-item` elements on the My Sites button. 
* * Remove `aria-haspopup` attribute from the 'My Sites' link to prevent it from triggering the wp-responsive-overlay on small screens (<=480 pixels). The responsive overlay would be blocking clicks inside of the expanded menu and should not be triggered. 

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






TBD
1. First let's verify the bug. Do this by going to any wp-admin page for a WordPress simple site and Atomic site.
2. Try to click on the My Sites button in the top bar navigation. You should be taken to the Calypso dashboard.
3. Now go back to wp-admin and resize the screen so that the menu disappears.
4. Click on the My Sites menu item and confirm that it still links to the Calypso dashboard.
5. Now it's time to apply the PR. You can follow the instructions in the comment below: https://github.com/Automattic/jetpack/pull/34807#issuecomment-1871926829 - **ALSO ensure that if you test with a simple site, that the url of your simple site is sandboxed (yoursite.wordpress.com should point to your sandbox)**
6. Once the PR has been applied to your sandbox and/or WoA dev site, navigate to wp-admin. If you applied the PR to your sandbox, you should go to wp-admin for a simple site. If you applied it to a WoA dev site, you should go to an Atomic site's wp-admin.
7. With a full screen, ensure that the sidebar is visible and try clicking on the 'My Sites' link again. Ensure that you still navigate to the Calypso dashboard.
8. Go back to wp-admin and resize the viewport until the sidebar is fully hidden. Now again click on the 'My Sites' button and confirm that the sidebar appears. Click on it again and confirm that the sidebar closes.
9. Finally, open the sidebar again by clicking on 'My Sites' and this time ensure that your viewport is scaled to the point that there is space on the right side of the sidebar. (there is a breakpoint where the expanded sidebar does not cover the entire screen). Click outside of the sidebar and confirm that the sidebar closes.

Ideally, these tests are done for both Atomic and WordPress simple sites. Also, check on your mobile phone and/or tablet. Testing in very small viewports is needed as there were some conflicts with the existing behaviour of the menu in relation to the `wp-responsive-overlay` being triggered on small screens. 

Viewports to test:
- `> 960` - Should always show the full menu with full text buttons. 
- `> 782 && <= 960` - Should always show the menu but only with icons.
- `> 480 && <= 782` - Should collapse the menu and it should be possible to toggle it via 'My Sites' button in top menu bar.
- `<= 480` - Should collapse the menu and it should be possible to toggle the menu via the WordPress logo in the top menu bar.

Please test the full menu experience to ensure there are no quirks and it works as you would expect.
